File: js/나이트의 이동(7562).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const T = Number(input[0]);
let index = 1;

for (let t=0; t < T; t++) {
    const L = Number(input[index]);
    index++;

    const [start_x, start_y] = input[index].split(' ').map(Number);
    index++;

    const [end_x, end_y] = input[index].split(' ').map(Number);
    index++;

    const arr = Array.from({length: L}, () => Array(L).fill(0));

    const dx = [-1, -2, -2, -1, 1, 2, 2, 1];
    const dy = [-2, -1, 1, 2, 2, 1, -1, -2];

    function bfs(x,y) {
        const queue = [];
        queue.push([x,y]);
        arr[x][y] = 1;

        while (queue.length) {
            const [x,y] = queue.shift();

            if (x === end_x && y === end_y) {
                console.log(arr[x][y]-1);
                return;
            }

            for (let i=0; i<8; i++) {
                const nx = x + dx[i];
                const ny = y + dy[i];

                if (nx < 0 || nx >= L || ny < 0 || ny >= L) continue;

                if (arr[nx][ny] === 0) {
                    arr[nx][ny] = arr[x][y] + 1;
                    queue.push([nx,ny]);
                }
            }
        }
    }
    bfs(start_x, start_y);
}

File: python/나이트의 이동(7562).py
from collections import deque

T = int(input())
for _ in range(T):
    L = int(input())
    arr = [[0] * L for _ in range(L)]

    start_x, start_y = map(int, input().split())
    end_x, end_y = map(int, input().split())

    dx = [-1, -2, -2, -1, 1, 2, 2, 1]
    dy = [-2, -1, 1, 2, 2, 1, -1, -2]

    def dfs(x,y):
        queue = deque()
        queue.append([x,y])
        arr[x][y] = 1
        while queue:
            x, y = queue.popleft()

            if x == end_x and y == end_y:
                    print(arr[x][y]-1)
                    return

            for i in range(8):
                nx = x + dx[i]
                ny = y + dy[i]

                if nx < 0 or nx >= L or ny < 0 or ny >= L:
                    continue

                if arr[nx][ny] == 0:
                    arr[nx][ny] = arr[x][y] + 1
                    queue.append([nx,ny])
    dfs(start_x, start_y)